crates/ra_syntax/src/grammar/types.rs

```diff
@@ -36,6 +36,14 @@ fn type_with_bounds_cond(p: &mut Parser, allow_bounds: bool) {
     }
 }
 
+pub(super) fn is_type_start(p: &mut Parser) -> bool {
```

Comment:
We probably can use https://github.com/rust-analyzer/rust-analyzer/blob/e396cb3548c8a36ba505cc08248333c225e56d2a/crates/ra_syntax/src/grammar/types.rs#L3 here.

`is_type_start(p)` should be replaced with `p.at_ts(TYPE_FIRST)`.

Reply:
Wont this fail if the pred is a path starting with `crate` or `super`? I don't see those tokens in `TYPE_FIRST`.

Reply:
TYPE_FIRST includes PATH_FIRST: `paths::PATH_FIRST.union`. That includes thouse tokens:

https://github.com/rust-analyzer/rust-analyzer/blob/e396cb3548c8a36ba505cc08248333c225e56d2a/crates/ra_syntax/src/grammar/paths.rs#L4

Reply:
Aha, ofcourse